docs/core/extensions/windows-service-with-installer.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Create a Windows Service installer
+description: Learn how to create a Windows Service installer project.
+author: IEvangelist
+ms.author: dapine
+ms.date: 02/15/2023
+ms.topic: tutorial
+---
+
+# Create a Windows Service installer
+
+When you create a .NET Windows Service (not to be mistaken with a .NET Framework Windows Service), you may want to create an installer for your service. Without an installer, users would have to know how to install and configure your service. An installer bundles your app's executables and exposes a customizable installation user experience.
+
+In this tutorial, you'll learn how to:
+
+> [!div class="checklist"]
+>
+> - Install the Visual Studio Installer Projects extension.
+> - Create a Setup Project.
+> - Update an existing .NET Worker project to support installation.
+> - Automate the installation and uninstallation with the Windows Service Control Manager.
+
+## Prerequisites
+
+- The [.NET 6.0 SDK or later](https://dotnet.microsoft.com/download/dotnet)
+- A Windows OS
+- A .NET integrated development environment (IDE)
+  - Feel free to use [Visual Studio](https://visualstudio.microsoft.com)
+- An existing .NET Windows Service
+
+## Install the extension
+
+Install the [Microsoft Visual Studio Installer Projects extension](https://marketplace.visualstudio.com/items?itemName=VisualStudioClient.MicrosoftVisualStudio2022InstallerProjects). After installing, restart Visual Studio and you'll see new project templates available.
+
+## Update existing project
+
+This tutorial is based on the app created as part of the [Create a Windows Service using `BackgroundService`](windows-service.md) tutorial. You can either clone the sample repo or use the app you built in the previous tutorial.
+
+[!INCLUDE [workers-samples-browser](includes/workers-samples-browser.md)]
+
+Open the solution in Visual Studio, and select <kbd>F5</kbd> to ensure that the app builds and runs as expected. Press <kbd>Ctrl</kbd>+<kbd>C</kbd> to stop the app.
+
+### Handle installation switches
+
+The Windows Service app needs to handle installation switches. The Setup Project will call into the Windows Service app with `/Install` and `/Uninstall` switches that are called during installation and uninstallation respectively. When these switches are present the app will behave differently, in that it will only perform installation or uninstallation using the Windows Service Control Manager executable (_sc.exe_).
```

Comment:
```suggestion
The Windows Service app needs to handle installation switches. The setup project will call into the Windows Service app with `/Install` and `/Uninstall` switches during installation and uninstallation respectively. When these switches are present, the app will behave differently, in that it will only perform installation or uninstallation using the Windows Service Control Manager executable (_sc.exe_).
```